lib/spack/spack/analyzers/libabigail.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+import spack
+import spack.error
+import spack.bootstrap
+import spack.binary_distribution
+import spack.package
+import spack.repo
+
+import llnl.util.tty as tty
+
+from .analyzerbase import Analyzerbase
+
+import os
+
+
+class Libabigail(Analyzerbase):
+
+    name = "libabigail"
+    outfile = "spack-analyzer-libabigail.json"
+    description = "Application Binary Interface (ABI) features for objects"
+
+    def __init__(self, spec, dirname=None):
+        """init for an analyzer is where we want to ensure that we have all
+        needed dependencies. For the abigail analyzer, this means Libabigail.
+        Since the output for libabigail is one file per object, we communicate
+        with the monitor multiple times.
+        """
+        super(Libabigail, self).__init__(spec, dirname)
+
+        # This doesn't seem to work to import on the module level
```

Comment:
For most of this stuff, I think we do not want to import at module level, since if it is an optional dependency it should only be bootstrapped *if* we need the functionality.  If you import at module level, it'll always be bootstrapped if you import the analyzer, and I think we want to be lazier than that.

Reply:
So this implementation is correct?

Reply:
LGTM!